keras_cv/layers/preprocessing/random_crop.py

```diff
@@ -80,14 +81,47 @@ def augment_image(self, image, transformation, **kwargs):
         )
 
     def compute_image_signature(self, images):
-        output_image_shape = list(tf.shape(images))
-        output_image_shape[H_AXIS] = self.height
-        output_image_shape[W_AXIS] = self.width
         return tf.TensorSpec(
-            shape=output_image_shape,
+            shape=(self.height, self.width, images.shape[-1]),
             dtype=self.compute_dtype,
         )
 
+    def augment_bounding_boxes(
+        self, bounding_boxes, transformation, image=None, **kwargs
+    ):
+        if self.bounding_box_format is None:
+            raise ValueError(
+                "`RandomCrop()` was called with bounding boxes,"
+                "but no `bounding_box_format` was specified in the constructor."
+                "Please specify a bounding box format in the constructor. i.e."
+                "`RandomCrop(bounding_box_format='xyxy')`"
+            )
+        bounding_boxes = bounding_box.convert_format(
+            bounding_boxes,
+            source=self.bounding_box_format,
+            target="xyxy",
+            images=image,
+        )
+        image_shape = tf.shape(image)
+        h_diff = image_shape[H_AXIS] - self.height
+        w_diff = image_shape[W_AXIS] - self.width
+        bounding_boxes = tf.cond(
+            tf.reduce_all((h_diff >= 0, w_diff >= 0)),
+            lambda: self._crop_bounding_boxes(image, bounding_boxes, transformation),
+            lambda: self._resize_bounding_boxes(
```

Comment:
Are we always resizing down? If not, we may need to clip these to the image (using keras_cv.bounding_box.clip_to_image)

Reply:
Ahh! just figured I can provide image_shape to clip_to_image. I was stuck here thinking it only crops to original image size.
I have updated the changes.